src/fractals/newtons_method.rs

```diff
@@ -0,0 +1,372 @@
+                        );
+                        rgb[i] = color.clamp(0.0, 255.0).round() as u8;
+                    }
+
+                    ColorMapKeyFrame {
+                        query: keyframe.query,
+                        rgb_raw: rgb,
+                    }
+                })
+                .collect();
+            inner_color_maps.push(ColorMap::new(&keyframes, LinearInterpolator));
+            color_maps.push(ColorMapLookUpTable::from_color_map(
+                inner_color_maps.last().unwrap(),
+                params.lookup_table_count,
+            ))
+        }
+        let histogram = Histogram::new(
+            params.histogram_bin_count,
+            params.max_iteration_count as f32,
+        );
+        let mut renderable = Self {
+            system,
+            cdf: CumulativeDistributionFunction::new(&histogram),
+            histogram: histogram.into(),
+            color_maps,
+            inner_color_maps,
+            params,
+        };
+        renderable.update_color_map();
+        renderable
+    }
+
+    fn newton_rhapson_iteration_sequence(&self, z0: Complex64) -> Option<NewtonRhapsonResult> {
+        newton_rhapson_iteration_sequence(
+            &self.system,
+            z0,
+            self.params.convergence_tolerance,
+            self.params.max_iteration_count,
+        )
+    }
+
+    fn update_color_map(&mut self) {
+        // This histogram uses data shared from all roots, so we do not need the `_soln` value in the below
+        // closure. Then we update all color maps based on the shared CDF, which is generated from the histogram.
+        populate_histogram(
+            &|point: &[f64; 2]| {
+                self.newton_rhapson_iteration_sequence(Complex64::new(point[0], point[1]))
+                    .map(|result| result.iteration_count as f32)
+            },
+            &self.params.image_specification,
+            self.params.histogram_bin_count as u32,
+            self.histogram.clone(),
+        );
+        self.cdf.reset(&self.histogram);
+
+        for (color_table, inner_map) in self.color_maps.iter_mut().zip(self.inner_color_maps.iter())
+        {
+            reset_color_map_lookup_table_from_cdf(color_table, &self.cdf, inner_map);
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum SystemType {
+    RootsOfUnity(Box<RootsOfUnityParams>), // number of roots == root_colors_rgb.len()
+}
+
+impl<F> SpeedOptimizer for NewtonsMethodRenderable<F>
+where
+    F: ComplexFunctionWithSlope,
+{
+    type ReferenceCache = CommonParams;
+    fn reference_cache(&self) -> CommonParams {
+        self.params.clone()
+    }
+
+    fn set_speed_optimization_level(&mut self, level: f64, cache: &Self::ReferenceCache) {
+        self.params
+            .render_options
+            .set_speed_optimization_level(level, &cache.render_options);
+
+        // TODO:  finish implementing this method to improve responsive performance in explore.
```

Comment:
I'll add this in a follow-up. I started working on this and found some other issues in the implementation of this trait for other fractals (DDP seems to have some of the scaling inverted...), so I want to do an "across the board fix" while I'm at it.